src/ride/controllers/update_ride_controller.py

```diff
@@ -0,0 +1,66 @@
+from api.utils import failure_response
+from api.utils import success_response
+from api.utils import update
+from django.core.exceptions import ObjectDoesNotExist
+from person.models import Person
+
+from ..models import Ride
+
+
+class UpdateRideController:
+    def __init__(self, request, data, serializer, id):
+        self._request = request
+        self._data = data
+        self._serializer = serializer
+        self._id = id
+
+    def process(self):
+        if not Ride.objects.filter(path_id=int(self._id)).exists():
+            return failure_response("Ride does not exist")
+        ride = Ride.objects.get(path_id=self._id)
```

Comment:
why is the ID I am passing into a ride endpoint being used to get a ride based on the path_id? I understand that it's a one-to-one relationship but semantically it makes sense that you get the ride based on it's own ID and then you can extract the path from it. Worst case scenario, the IDs for rides and paths do not match (for whatever reason) and then your update ride controller starts updating the wrong ride.